### PR TITLE
[INTEL MKL] Changes to oneDNN build to remove binary blob when building with opensource components only.

### PR DESCRIPTION
**Pull request by agramesh1**
The PR includes the following changes.
1) Changes to the mkl_threapool build so it doesn't depend on the MKL binary blob.
2) Added  the mkl_opensource_only config option that enables building of TF+oneDNN without the MKL binary blob.
3) Cleaning up obsolete code.

### CODE REVIEW

File: tensorflow/core/kernels/mkl_batch_matmul_op.cc [hunk 1]
```diff
@@ -15,21 +15,26 @@ limitations under the License.
...
+// Batch Matrix-Matrix Multiplication (MatMul) operations.
+// We currently register this kernel only for oneDNN supported data
+// types (float, bfloat16). This file can be built with and without the use of
+// the binary MKL CBLAS calls, controlled by the macro INTEL_MKL_DNN_ONLY.
+// If INTEL_MKL_DNN_ONLY is defined, only oneDNN is used. For cases not
+// supported by oneDNN (ex. Batchmatmul with broadcasting) we fall back to the
+// default CPU implementation.
+// if INTEL_MKL_DNN_ONLY is not defined, both oneDNN and MKL CBLAS
+// implementations are used. This is only temporary, once we are able handle all
+// cases with oneDNN, CBLAS calls will be removed.
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_batch_matmul_op.cc:28]: Thank you for the detailed documentation!

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! The new configurations make it less confusing. 👍 
I have two minor nits please.

**Review by penpornk — DISMISSED** [review]
Thank you!

**Comment by agramesh1** [comment]
> @agramesh1 Can you please check build failures. Thanks!

@gbaned thank you, I will take a look at it.

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Review by rthadur — APPROVED** [review]

**Pull request merged**